tests/Unit/Evolution/DgSubcell/Actions/Test_ReconstructionCommunication.cpp

```diff
@@ -609,12 +653,454 @@ void test(const bool use_cell_centered_flux) {
+    const evolution::dg::subcell::RdmpTciData& rdmp_tci_data) {
+  const size_t rdmp_max_values_size = rdmp_tci_data.max_variables_values.size();
+
+  // Copy max_variables_values after the original data
+  std::copy(
+      rdmp_tci_data.max_variables_values.cbegin(),
+      rdmp_tci_data.max_variables_values.cend(),
+      data_to_fill.begin() + static_cast<std::ptrdiff_t>(original_data_size));
+
+  // Copy min_variables_values after max_variables_values
+  std::copy(
+      rdmp_tci_data.min_variables_values.cbegin(),
+      rdmp_tci_data.min_variables_values.cend(),
+      data_to_fill.begin() + static_cast<std::ptrdiff_t>(original_data_size +
+                                                         rdmp_max_values_size));
+}
+
+template <size_t Dim, bool UseNodegroupDgElements, typename MetaVars>
+DataVector compute_expected_ghost_data(
+    const ActionTesting::MockRuntimeSystem<MetaVars>& runner,
+    const Element<Dim>& element, const Direction<Dim>& direction,
+    const ::Mesh<3> subcell_mesh, const bool enable_extension,
+    const bool use_cell_centered_flux,
+    const Variables<tmpl::list<Var1>>& evolved_vars,
+    const evolution::dg::subcell::RdmpTciData& rdmp_tci_data,
+    const TimeStepId time_step_id, const size_t ghost_zone_size) {
+  using comp = component<Dim, MetaVars, UseNodegroupDgElements, true>;
+  using CellCenteredFluxTag =
+      evolution::dg::subcell::Tags::CellCenteredFlux<tmpl::list<Var1>, Dim>;
+  using ActionTesting::get_databox_tag;
+  const auto& element_id = element.id();
+  const auto& fd_to_neighbor_fd_interpolants = get_databox_tag<
+      comp, evolution::dg::subcell::Tags::InterpolatorsFromFdToNeighborFd<Dim>>(
+      runner, element_id);
+  const auto& extension_directions = get_databox_tag<
+      comp, evolution::dg::subcell::Tags::ExtensionDirections<Dim>>(runner,
+                                                                    element_id);
+  const size_t rdmp_size = rdmp_tci_data.max_variables_values.size() +
+                           rdmp_tci_data.min_variables_values.size();
+
+  if (enable_extension and extension_directions.contains(direction)) {
+    const auto& receiver_id = element.neighbors().at(direction).ids().begin();
+    const auto& extender_direction =
+        extension_directions.at(direction).direction_to_extend;
+    const auto& extender_id =
+        element.neighbors().at(extender_direction).ids().begin();
+    const auto& data_from_extender =
+        ActionTesting::get_inbox_tag<
+            comp, evolution::dg::Tags::BoundaryCorrectionAndGhostCellsInbox<
+                      Dim, UseNodegroupDgElements>>(runner, element_id)
+            .at(time_step_id)
+            .at(DirectionalId<Dim>{extender_direction, *extender_id})
+            .ghost_cell_data.value();
+
+    DataVector relevant_ghost_data;
+    relevant_ghost_data.set_data_ref(
+        const_cast<double*>(data_from_extender.data()),  // NOLINT
```

Comment:
same comment about `NOLINT`. We also have a function in `VectorImpl.hpp` that can create a const view. Make `relevant_ghost_data` const and then use that

Reply:
done. also changed to using make_const_view for the similar line in `ReconstructionCommunication.hpp`